src/optimizer_interface.jl

```diff
@@ -123,26 +123,44 @@ function set_optimizer(
 end
 
 """
-    optimize!(model::Model;
-              ignore_optimize_hook=(model.optimize_hook === nothing),
-              kwargs...)
+    optimize!(
+        model::Model;
+        ignore_optimize_hook = (model.optimize_hook === nothing),
+        differentiation_backend::MOI.Nonlinear.AbstractAutomaticDifferentiation =
+            MOI.Nonlinear.SparseReverseMode(),
+        kwargs...,
+    )
+
+Optimize the model.
 
-Optimize the model. If an optimizer has not been set yet (see
-[`set_optimizer`](@ref)), a [`NoOptimizer`](@ref) error is thrown.
+If an optimizer has not been set yet (see [`set_optimizer`](@ref)), a
+[`NoOptimizer`](@ref) error is thrown.
 
-Keyword arguments `kwargs` are passed to the `optimize_hook`. An error is
-thrown if `optimize_hook` is `nothing` and keyword arguments are provided.
+If `ignore_optimize_hook == true`, the optimize hook is ignored and the model is
+solved as if the hook was not set. Keyword arguments `kwargs` are passed to the
+`optimize_hook`. An error is thrown if `optimize_hook` is `nothing` and keyword
+arguments are provided.
+
+Pass `differentiation_backend` to set the `AbstractAutomaticDifferentiation`
```

Comment:
Should we add a link to MOI's documentation there?

Reply:
Done